realm/src/main/java/io/realm/Realm.java

```diff
@@ -317,12 +320,14 @@ public void setAutoRefresh(boolean autoRefresh) {
 
     // Public because of migrations
     public Table getTable(Class<?> clazz) {
-        String simpleClassName = simpleClassNames.get(clazz);
-        if (simpleClassName == null) {
-            simpleClassName = clazz.getSimpleName();
-            simpleClassNames.put(clazz, simpleClassName);
+        final String proxySuffix = "RealmProxy";
+        String proxiedClassName = proxiedClassNames.get(clazz);
+        if (proxiedClassName == null) {
+            String classSimpleName = clazz.getSimpleName();
+            proxiedClassName = classSimpleName.replace(proxySuffix, "");
```

Comment:
you are correct, but please notice it's called proxiedClassNames, not proxyClassNames. Meaning it's the name of the class being proxied.